data/cookbook/web-programming/uri-encoding/00-uri.ml

```diff
@@ -0,0 +1,42 @@
+---
+packages:
+- name: "uri"
+  version: "4.2.0"
+libraries:
+  - uri
+discussion: |
+  - **Understanding `Uri`:** `Uri` provides useful functions for parsing and encoding URI (URL or URN). It can be handy with complex URI (parameters, special or non ASCII characters)
```

Comment:
```suggestion
  - **Understanding `Uri`:** `Uri` provides useful functions for parsing and encoding URI (URL or URN). It can be handy with complex URI (parameters, special, or non-ASCII characters)
```

This serial comma is needed if there are three things here: parameters, special, and non-ASCII characters). However, if you mean "parameters," and two examples of parameters are "special and non-ASCII characters, then the punctuation is correct.